service/utils/channel_utils.js

```diff
@@ -98,7 +98,8 @@ function isDirectChannelVisible(userId, myPreferences, channel) {
 
 function isFavoriteChannel(myPreferences, channel) {
     const fav = myPreferences[`${Constants.CATEGORY_FAVORITE_CHANNEL}--${channel.id}`];
-    return fav && fav.value === 'true';
+    channel.isFavorite = fav && fav.value === 'true';
```

Comment:
actually if you take a look at the selector i'm doing basically the same thing with the `isCurrent` property, the store is not being mutated, just the object that is being used which has already been copied here https://github.com/mattermost/mattermost-mobile/blob/master/service/selectors/entities/channels.js#L60